src/raftstore/store/peer.rs

```diff
@@ -1941,12 +1958,29 @@ impl Peer {
             }
         }
 
+        // When a replica cannot detect any leader, `MsgReadIndex` will be dropped, which would
+        // cause a long time waiting for a read response. Then we should return an error directly
+        // in this situation.
+        if !self.is_leader() && self.leader_id() == INVALID_ID {
+            cmd_resp::bind_error(
```

Comment:
maybe update related metrics?

Reply:
> maybe update related metrics?

Should it be a type of `unsafe_read_index`?

Reply:
not for sure, add a new one? @overvenus What's your opinion?

Reply:
What about `no_leader` or `follower_read_no_leader`? You can add it to `RaftInvalidProposeMetrics`.

Reply:
OK, I will add a `read_index_no_leader` there.